pkg/monitoring/domainstats/downwardmetrics/hostmetrics.go

```diff
@@ -1,69 +1,115 @@
 package downwardmetrics
 
 import (
+	"bufio"
+	"fmt"
 	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
 	"time"
 
-	"github.com/c9s/goprocinfo/linux"
-
+	"github.com/prometheus/procfs"
+	"github.com/prometheus/procfs/sysfs"
 	"kubevirt.io/client-go/log"
 
 	"kubevirt.io/kubevirt/pkg/downwardmetrics/vhostmd/api"
 	metricspkg "kubevirt.io/kubevirt/pkg/downwardmetrics/vhostmd/metrics"
 )
 
 type hostMetricsCollector struct {
-	procCPUInfo string
-	procStat    string
-	procMemInfo string
-	procVMStat  string
-	pageSize    int
+	procPath string
+	sysPath  string
+	pageSize int
 }
 
-func (h *hostMetricsCollector) hostCPUMetrics() (metrics []api.Metric) {
-	if cpuinfo, err := linux.ReadCPUInfo(h.procCPUInfo); err == nil {
-		metrics = append(metrics,
-			metricspkg.MustToUnitlessHostMetric(cpuinfo.NumPhysicalCPU(), "NumberOfPhysicalCPUs"),
-		)
-	} else {
-		log.Log.Reason(err).Info("failed to collect cpuinfo on the node")
-	}
+func (h *hostMetricsCollector) hostCPUMetrics() []api.Metric {
+	var metrics []api.Metric
+	metrics = append(metrics, h.hostPhysicalCpuCount()...)
 
-	if stat, err := linux.ReadStat(h.procStat); err == nil {
-		// CLK_TCK is a constant on Linux, see e.g.
-		// https://github.com/containerd/cgroups/pull/12
-		var clk_tck float64 = 100
-		cpuTime := float64(stat.CPUStatAll.User+stat.CPUStatAll.Nice+stat.CPUStatAll.System) / clk_tck
+	procFS, err := procfs.NewFS(h.procPath)
+	if err != nil {
+		log.Log.Reason(err).Info("failed to access /proc")
+		return nil
+	}
+	if stat, err := procFS.Stat(); err == nil {
+		cpuTime := stat.CPUTotal.User + stat.CPUTotal.Nice + stat.CPUTotal.System
 		metrics = append(metrics,
 			metricspkg.MustToHostMetric(cpuTime, "TotalCPUTime", "s"),
 		)
 	} else {
 		log.Log.Reason(err).Info("failed to collect cputime on the node")
 	}
```

Comment:
maybe this can be also improved to:

```suggestion
	
	stat, err := procFS.Stat()
	if err != nil {
	        log.Log.Reason(err).Info("failed to collect cputime on the node")
	        return nil
	}
	
	cpuTime := stat.CPUTotal.User + stat.CPUTotal.Nice + stat.CPUTotal.System
	metrics = append(metrics, metricspkg.MustToHostMetric(cpuTime, "TotalCPUTime", "s"))
```